WeakAuras.lua

```diff
@@ -3138,9 +2980,268 @@ function WeakAuras.RegisterTriggerSystem(types, triggerSystem)
+          function()
+            state.show = false;
+            state.changed = true;
+            WeakAuras.UpdatedTriggerState(id);
+          end,
+          state.duration);
+        record.expirationTime = state.expirationTime;
+      end
+    else -- no auto hide, delete timer
+      if (timers[id] and timers[id][triggernum] and timers[id][triggernum][cloneId]) then
+        local record = timers[id][triggernum][cloneId];
+        if (record.handle) then
+          timer:CancelTimer(record.handle);
+        end
+        record.handle = nil;
+        record.expirationTime = nil;
+      end
+    end
+  else -- not shown
+    if(timers[id] and timers[id][triggernum] and timers[id][triggernum][cloneId]) then
+      local record = timers[id][triggernum][cloneId];
+      if (record.handle) then
+        timer:CancelTimer(record.handle);
+      end
+      record.handle = nil;
+      record.expirationTime = nil;
+    end
+  end
+end
+
+local function ApplyStateToRegion(id, region, state)
+  region.state = state;
+  if(region.SetDurationInfo) then
+    if (state.progressType == "timed") then
+      local now = GetTime();
+      local value = math.huge - now;
+      if (state.expirationTime and state.expirationTime > 0) then
+        value = state.expirationTime - now;
+      end
+      local total = state.duration or 0
+      local func = nil;
+
+      region:SetDurationInfo(total,
+                             now + value,
+                             func,
+                             state.inverseDirection);
+    elseif (state.progressType == "static") then
+      local value = state.value or 0;
+      local total = state.total or 0;
+      local durationFunc = state.durationFunc or true;
+
+      region:SetDurationInfo(value, total, durationFunc or true, state.inverseDirection);
+    else
+      -- Do nothing, should ideally clear duration info on region
+    end
+  end
+  if (state.resort) then
+    WeakAuras.ControlChildren(region.id);
+    state.resort = false;
+  end
+  if(region.SetName) then
+    region:SetName(state.name);
+  end
+  if(region.SetIcon) then
+    region:SetIcon(state.icon or "Interface\\Icons\\INV_Misc_QuestionMark");
+  end
+  if(region.SetStacks) then
+    region:SetStacks(state.stacks);
+  end
+  if(region.UpdateCustomText and not WeakAuras.IsRegisteredForCustomTextUpdates(region)) then
+    region.UpdateCustomText();
+  end
+  WeakAuras.UpdateMouseoverTooltip(region);
+  region:Expand();
+end
+
+-- Fallbacks if the states are empty
+local emptyTable = {}
+local emptyState = {};
+emptyState[""] = emptyTable;
+
+local function applyToTriggerStateTriggers(stateShown, id, triggernum)
+  if (stateShown and not triggerState[id].triggers[triggernum + 1]) then
+    triggerState[id].triggers[triggernum + 1] = true;
+    triggerState[id].triggerCount = triggerState[id].triggerCount + 1;
+    return true;
+  elseif (not stateShown and triggerState[id].triggers[triggernum + 1]) then
+    -- Check if any other clone is shown
```

Comment:
We are hiding a state for the trigger x, but there could be clones that are visible. In that case the trigger x is still active.